craft_parts/infos.py

```diff
@@ -269,6 +270,24 @@ def partitions(self) -> list[str] | None:
         """Return the project's partitions."""
         return self._partitions
 
+    @property
+    def default_partition(self) -> str | None:
+        """Get the "default" partition from a partition list."""
+        if self._partitions:
+            return self._partitions[0]
+        return None
+
+    def is_default_partition(self, partition: str | None) -> bool:
+        """Check if given partition is the default one."""
+        if self._partitions is None and (
+            partition is None or partition == DEFAULT_PARTITION
+        ):
+            return True
+        if self._partitions is not None:
+            return partition == self._partitions[0]
+
+        return False
```

Comment:
These are sadly slightly different to accommodate their respective usage and due to the handling of the `None`/`"default"` values when partitions are defined or not.